src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -820,6 +805,48 @@ public void setWebProxy(WebProxy value) {
   public Map<String, String> getHttpHeaders() {
     return this.httpHeaders;
   }
+  
+  protected final CloseableHttpClient getHttpClient(){
```

Comment:
How to configure/use custom httpClient or poolingHttpClient (with some additional configuration)? Maybe it will be better to move some part of code in factory methods?